app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,28 @@
+class VotesController < ApplicationController
+  def create
+    new_vote = params[:newVote]
+    r_id = params[:rId].to_i
+
+    vote = Vote.find_by(user: current_user, review_id: r_id)
+
+    if vote
+      if vote.upvote == (new_vote == 'upvote')
+        vote.destroy
+      else
+        vote.upvote = !vote.upvote
+        vote.save
+      end
+    else
+      Vote.create(
+        user: current_user,
+        review_id: r_id,
+        upvote: new_vote == 'upvote'
+      )
+    end
```

Comment:
This looks good, you may be able to refactor a bit to use a method that comes with active record `.find_or_create_by()`http://guides.rubyonrails.org/active_record_querying.html#find-or-create-by